restart-script/restart-cluster

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env bash
+
+red='\033[1;31m'
+grn='\033[1;32m'
+yel='\033[1;33m'
+blu='\033[1;36m'
+pnk='\033[1;35m'
+clr='\033[0m'
+
+READLINK=`readlink $0`
+if [[ "$READLINK" == "" ]]; then DIR="$PWD"; else DIR="$(dirname $READLINK)"; fi
+
+checkHosts() {
+  if [ -z "${HOSTS_FILE:-}" ]; then
+    echo -e "${red}HOSTS_FILE was not set${clr}"
+    exit 1
+  fi
+}
+
+setHosts() {
+  HOSTS_FILE=$1
+}
+
+sshGenesis() {
+  checkHosts
+  genesis_ip=`head -1 $HOSTS_FILE`
+  ssh admin@$genesis_ip -o StrictHostKeyChecking=no -n $1
+}
+
+checkNodeState() {
+  node_to_check=$1
+  port_to_check=$2
+  expected_node_state=$3
+  local actual_node_state
+
+  while [ "$expected_node_state" != "$actual_node_state" ]; do
+          actual_node_state=$(sshGenesis "curl -s http://$node_to_check:$port_to_check/cluster/info | jq '.[] | select( .ip == \"$node_to_check\" )' | jq '.state' -r")
+    echo -e "${pnk} Current node ($node_to_check) state: $actual_node_state. Expected state: $expected_node_state ${clr}"
+    sleep 1
+  done
+}
+
+stopServices() {
+  echo -e "${blu}Stopping node snd dag services${clr}"
+
+  while read -r ip; do
+    echo "Stopping services for: $ip"
+    ssh admin@$ip -o StrictHostKeyChecking=no -n 'sudo systemctl stop dag && sudo systemctl stop node'
+  done <$HOSTS_FILE
+}
+
+downloadRollbackSnapshot() {
+  echo -e "${blu}Downloading latest snapshot${clr}"
+
+  sshGenesis 'curl -s https://olubc3p4rc.execute-api.us-west-1.amazonaws.com/global-snapshots/latest | jq '.data.hash' -r > /home/admin/tessellation/rollback-snapshot'
```

Comment:
what's that url? if this is a testnet block explorer, it should be the domain name. I'd assign it to the variable at the beginning of the file btw